src/pool/srv_target.c

```diff
@@ -189,18 +190,16 @@ pool_child_add_one(void *varg)
 	rc = ds_mgmt_tgt_file(arg->pla_uuid, VOS_FILE, &info->dmi_tgt_id,
 			      &path);
 	if (rc != 0) {
-		D_FREE(child);
-		return rc;
+		D_FREE(path);
+		goto out_free;
```

Comment:
Why freeing path when rc != 0? I think the existing code (without this patch) is correct.

Reply:
Fixed in #6200